graphium/nn/architectures/global_architectures.py

```
@@ -1278,6 +1301,95 @@ def set_max_num_nodes_edges_per_graph(self, max_nodes: Optional[int], max_edges:
+        task = cfg_finetune['task']
+        added_depth = cfg_finetune['added_depth']
+
+        for module in ['pre_nn', 'pre_nn_edges', 'gnn', 'graph_output_nn', 'task_heads']:
+
+            if module == cfg_finetune['module_from_pretrained']:
+                break
+
+            self.overwrite_complete_module(module, pretrained_model)
+
+        self.overwrite_partial_module(module, task, task_head_from_pretrained, added_depth, pretrained_model)       
+
+    def overwrite_partial_module(self, module, task, task_head_from_pretrained, added_depth, pretrained_model):
+        """Completely overwrite the specified module
+        """
+        if module == 'gnn':
+            shared_depth = len(self.task_heads.task_heads[task].layers) - added_depth
+            assert shared_depth >= 0
+            if shared_depth > 0:
+                self.gnn.layers[:shared_depth] = pretrained_model.gnn.layers[:shared_depth]
+
+        if module == "task_heads":
+            shared_depth = len(self.task_heads.task_heads[task].layers) - added_depth
+            assert shared_depth >= 0
+            if shared_depth > 0:
+                self.task_heads.task_heads[task].layers = pretrained_model.task_heads.task_heads[task_head_from_pretrained].layers[:shared_depth] + self.task_heads.task_heads[task].layers[shared_depth:]
+
+        elif module in ['pre_nn', 'pre_nn_edges']:
+            raise NotImplementedError(f"It is unlikely that we will want to finetune from pre-NNs")
+        
+        else:
+            raise NotImplementedError()
+
+    def overwrite_complete_module(self, module, pretrained_model):
+        """Completely overwrite the specified module
+        """
+        if module == "pre_nn":
+            try:
+                self.pre_nn.layers = pretrained_model.pre_nn.layers
+            except:
+                logger.warning(f"Pretrained ({pretrained_model.pre_nn}) and/or finetune model ({self.pre_nn}) do not use a pre-NN.")
+                pass
```

Just a warning? This seems like a major configuration error and should throw an Exception? 

(FYI:  `pass` is not needed here)

---

Similar as above, this was part of finetuning v1, which we are not using. I have not removed the function because a similar function was already implemented by someone else (although it was not used anywhere).

Do you recommend removing the part of the code?